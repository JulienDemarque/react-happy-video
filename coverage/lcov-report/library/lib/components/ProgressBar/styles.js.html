<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for library/lib/components/ProgressBar/styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">library/lib/components/ProgressBar</a> styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components';
import { lighten, tint } from 'polished';
import PropTypes from 'prop-types';
&nbsp;
export const Container = styled.div`
	width: 100%;
`;
&nbsp;
export const FullBar = styled.div`
	width: 100%;
	background-color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; {
<span class="cstat-no" title="statement not covered" >		return lighten(0.2, props.colorgiven);</span>
	}};
	height: 20px;
	margin: 0;
`;
&nbsp;
export const Progress = styled.div.attrs(<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
	style: { width: `${props.progress * 100}%` }
}))`
	background-color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; {
<span class="cstat-no" title="statement not covered" >		return tint(0.7, props.colorgiven);</span>
	}};
	height: 20px;
	margin: 0;
`;
&nbsp;
Progress.propTypes = {
	progress: PropTypes.number.isRequired
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Dec 04 2018 19:51:02 GMT+0200 (IST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
