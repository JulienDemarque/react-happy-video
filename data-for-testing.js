export const ProgressBarProps = {
	colorgiven: '#3b3346',
	progress: 0.2
};

export const SpeedControlProps = {
	speed: 1
};

export const TimeDisplayProps = {
	time: 20,
	duration: 300
};

export const VideoPlayerProps = {
	width: '100%',
	source: 'www.mymockurl.com/mymockvideo.mp4',
	color: '#3b3346'
};

export const VolumeControlProps = {
	volume: 0.5
};
