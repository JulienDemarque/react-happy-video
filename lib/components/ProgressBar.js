import React from 'react';
import PropTypes from 'prop-types';
import {
	FullBar,
	Progress,
	Container
} from '../styles/components/progress-bar';

//___________________________________________________________
// Progress Bar Component
class ProgressBar extends React.Component {
	constructor(props) {
		super(props);
		this.handleOnClick = this.handleOnClick.bind(this);
	}

	handleOnClick(event) {
		const nodeBar = event.currentTarget;
		const x = event.clientX - nodeBar.getBoundingClientRect().left;
		const width = nodeBar.offsetWidth;
		const progress = x / width;
		this.props.moveCurrentTime(progress);
	}

	render() {
		const { progress } = this.props;
		return (
			<Container>
				<FullBar onClick={this.handleOnClick}>
					<Progress progress={progress} />
				</FullBar>
			</Container>
		);
	}
}

ProgressBar.propTypes = {
	widthgiven: PropTypes.string.isRequired,
	progress: PropTypes.number.isRequired
};
export default ProgressBar;
