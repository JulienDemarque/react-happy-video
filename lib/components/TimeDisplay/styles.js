import styled from 'styled-components';

export const SubContainer = styled.div`
	display: flex;
	align-items: center;
	padding: 0 20px;
`;
