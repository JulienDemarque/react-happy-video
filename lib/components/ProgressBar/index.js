import React from 'react';
import PropTypes from 'prop-types';
import { FullBar, Progress, Container } from './styles';

//___________________________________________________________
// Progress Bar Component
class ProgressBar extends React.Component {
	constructor(props) {
		super(props);
		this.handleOnClick = this.handleOnClick.bind(this);
	}

	handleOnClick(event) {
		const nodeBar = event.currentTarget;
		const x = event.clientX - nodeBar.getBoundingClientRect().left;
		const width = nodeBar.offsetWidth;
		const progress = x / width;
		this.props.moveCurrentTime(progress);
	}

	render() {
		const { progress, colorgiven } = this.props;
		return (
			<Container>
				<FullBar colorgiven={colorgiven} onClick={this.handleOnClick}>
					<Progress colorgiven={colorgiven} progress={progress} />
				</FullBar>
			</Container>
		);
	}
}

ProgressBar.propTypes = {
	progress: PropTypes.number.isRequired
};
export default ProgressBar;
