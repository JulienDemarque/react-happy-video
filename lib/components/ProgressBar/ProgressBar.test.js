import React from 'react';
import { shallow } from 'enzyme';
import { ProgressBarProps } from '../../../data-for-testing';
import ProgressBar from './index';
import { FullBar } from './styles';
const { colorgiven, progress } = ProgressBarProps;

describe('ProgressBar Component', () => {
	let wrapper;
	const mockMoveCurrentTime = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
			<ProgressBar
				colorgiven={colorgiven}
				progress={progress}
				moveCurrentTime={mockMoveCurrentTime}
			/>
		);
	});

	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should call the mock mockMoveCurrentTime function on click', () => {
		const mockedEvent = {
			currentTarget: {
				offsetWidth: 50,
				getBoundingClientRect: function() {
					return { left: 10 };
				}
			},
			clientX: 20
		};
		wrapper.find(FullBar).simulate('click', mockedEvent);
		expect(mockMoveCurrentTime.mock.calls.length).toBe(1);
	});
});
