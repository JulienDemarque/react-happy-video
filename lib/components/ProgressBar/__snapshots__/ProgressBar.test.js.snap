// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    colorgiven="#3b3346"
    moveCurrentTime={[MockFunction]}
    progress={0.2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        colorgiven="#3b3346"
        onClick={[Function]}
      >
        <ForwardRef
          colorgiven="#3b3346"
          progress={0.2}
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          colorgiven="#3b3346"
          progress={0.2}
        />,
        "colorgiven": "#3b3346",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "colorgiven": "#3b3346",
          "progress": 0.2,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            [Function],
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
	background-color: ",
              [Function],
              ";
	height: 20px;
	margin: 0;
",
            ],
          },
          "displayName": "styled.div",
          "propTypes": Object {
            "progress": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
	width: 100%;
	background-color: ",
            [Function],
            ";
	height: 20px;
	margin: 0;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          colorgiven="#3b3346"
          onClick={[Function]}
        >
          <ForwardRef
            colorgiven="#3b3346"
            progress={0.2}
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            colorgiven="#3b3346"
            progress={0.2}
          />,
          "colorgiven": "#3b3346",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "colorgiven": "#3b3346",
            "progress": 0.2,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
	background-color: ",
                [Function],
                ";
	height: 20px;
	margin: 0;
",
              ],
            },
            "displayName": "styled.div",
            "propTypes": Object {
              "progress": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
	width: 100%;
	background-color: ",
              [Function],
              ";
	height: 20px;
	margin: 0;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	width: 100%;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
